### ISSUE TITLE
plmblast.py script doesn't work when following example 

### ISSUE THREAD

**Issue by seanrjohnson** [labstructbioinf/pLM-BLAST#23]
I'm trying to run pLM-BLAST following the example in the documentation. My query and database are both the same protein fasta file, which has about 11,000 sequences in it. 

```bash
python pLM-BLAST/embeddings.py start ../peptides.fasta pLM-blastDB -embedder pt --gpu -bs 0 --asdir
python pLM-BLAST/scripts/dbtofile.py pLM-blastDB #needed to hack it a bit.
python pLM-BLAST/embeddings.py start ../peptides.fasta query.pt --gpu -bs 0

# at this point I have a directory pLM-blastDB with a bunch of .emb files and emb.64,
# I also have query.pt. I want to do all vs all search of query.pt against pLM-blastDB (which actually are the same sequences)

python pLM-BLAST/scripts/plmblast.py pLM-blastDB query pLM-blast_hits.csv --use_chunks #Does not work!

#gives error:
# FileNotFoundError: could not find input file for `pLM-blastDB` expecting one of the extensions .csv, .p, .pkl, .fas or .fasta
```

What am I doing wrong here?

**Comment by seanrjohnson** [comment]
Looks like [Citugulia40](https://github.com/Citugulia40) and I are running into the same issues at the same time 🙂.

**Comment by staszekdh** [comment]
There seems to be a problem. It is under investigation by our team :) 

As for the inconsistent formats (the same sequences are stored differently for the database and the query) - we are also working on proving a single unified format.

**Comment by Argusmocny** [comment]
The problem is that `plmblast.py` requires database and query to be both files (`.csv`/`.fasta` etc and `.pt` or directory with `.pt`) with the same names. In your example you are creating `pLM-blastDB` as embedding storage so `plmblast.py` will expect  `pLM-blastDB.fasta`. Which is non intiutive - we will make it more robust and user friendly soon. 
Just copy 
```bash
../peptides.fasta pLM-blastDB.fasta
```

**Comment by Citugulia40** [comment]
It is working now. Thanks.

**Comment by seanrjohnson** [comment]
Somehow still not working for me. I deleted my embedding files (query and database), pulled the newest code from github, copied the source file as `pLM-blastDB.fasta` and `query.fasta`, and reran the commands. It runs for a while but then returns the error:

```
Traceback (most recent call last):
  File "pLM-blast/pLM-BLAST/scripts/plmblast.py", line 155, in <module>
    raise ValueError(f'''
ValueError:
                                        The length of the embedding and the query sequence are different:
                                         query index 3009 1256 != 1000
```

**Comment by seanrjohnson** [comment]
[peptides.txt](https://github.com/labstructbioinf/pLM-BLAST/files/13128250/peptides.txt)
I attached my `peptides.fasta` file, in case that's helpful (renamed to peptides.txt, so that github will accept it as an attachment).

**Comment by Citugulia40** [comment]
Hi, I think you may need to set `-t` option which is mentioned in previous issue opened by me.

**Comment by Citugulia40** [comment]
@Argusmocny, I am getting a new error after the end of the process which is 

`Traceback (most recent call last):
  File "/data2/ccitu/plmblast/../software/pLM-BLAST/scripts/plmblast.py", line 217, in <module>
    query_result = aln.postprocess.prepare_output(args, rows, qid, query_seqs[qid], db_df)
  File "/data2/ccitu/plmblast/../software/pLM-BLAST/alntools/postprocess/format.py", line 61, in prepare_output
    raise KeyError(f'missing {col} column in input dataframe')
KeyError: 'missing id column in input dataframe'`

**Comment by seanrjohnson** [comment]
I made it to the second progress bar in my `plmblast.py` call. and at 298/33633, I crashed with the following error:
```
concurrent.futures.process._RemoteTraceback:
"""
Traceback (most recent call last):
  File "python3.10/concurrent/futures/process.py", line 246, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "pLM-BLAST/alntools/base.py", line 108, in full_compare
    res = self.embedding_to_span(emb1, emb2)
  File "pLM-BLAST/alntools/base.py", line 73, in embedding_to_span
    paths = gather_all_paths(densitymap,
  File "pLM-BLAST/alntools/alignment.py", line 285, in gather_all_paths
    indices = border_argmaxpool(score_matrix, cutoff=minlen, factor=bfactor)
  File "pLM-BLAST/alntools/alignment.py", line 184, in border_argmaxpool
    assert cutoff*2 < (array.shape[0] + array.shape[1]), 'cutoff exeed array size'
AssertionError: cutoff exeed array size
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "pLM-BLAST/scripts/plmblast.py", line 194, in <module>
    res = job.result()
  File "python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
AssertionError: cutoff exeed array size

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "pLM-BLAST/scripts/plmblast.py", line 200, in <module>
    raise AssertionError('job not done', e)
AssertionError: ('job not done', AssertionError('cutoff exeed array size'))

```

**Comment by Argusmocny** [comment]
This is probably because, some of your `query` or `database` sequences are very short, probably in range of requested minimal alignemnt lenght or window_size - this will be fixed in https://github.com/labstructbioinf/pLM-BLAST/issues/24

**Comment by Argusmocny** [comment]
@seanrjohnson  update is live

**Comment by seanrjohnson** [comment]
@Argusmocny 
Now I'm getting this error. I'll try redoing the embedding and see if that helps.
```
pLM-blast/pLM-BLAST/alntools/numeric.py:179: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit'
decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html
#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit(fastmath=True, cache=True)
Traceback (most recent call last):
  File "pLM-blast/pLM-BLAST/scripts/plmblast.py", line 164, in <module>
    query_filedict = filtering_db(args, query_embs)
  File "pLM-blast/pLM-BLAST/scripts/plmblast.py", line 76, in filtering_db
    query_emb_chunkcs = [avg_pool1d(emb.unsqueeze(0), 16).squeeze().float() for emb in query_embs]
  File "pLM-blast/pLM-BLAST/scripts/plmblast.py", line 76, in <listcomp>
    query_emb_chunkcs = [avg_pool1d(emb.unsqueeze(0), 16).squeeze().float() for emb in query_embs]
RuntimeError: "avg_pool2d" not implemented for 'Half'
```

**Comment by kkaminski2137** [comment]
This happens when embeddings are saved as `torch.float16` and average pooling is not implemented for this type. I have fixed this, please pull new version and try again. Thanks a lot for your feedback :)

**Comment by Argusmocny** [comment]
@seanrjohnson  This happens because embeddings were saved as `torch.float16` and `avg_pool2d` requires `torch.float32`. Strange because this script passed test locally, I have fixed this, please pull new updates and try again. Thanks for your exhausting feedback :)

**Comment by seanrjohnson** [comment]
I had to change:
`query_emb_chunkcs = [avg_pool1d(emb.unsqueeze(0), 16).squeeze() for emb in query_embs]`
to 
`query_emb_chunkcs = [avg_pool1d(emb.unsqueeze(0), 16).squeeze() for emb in query_emb_chunkcs]` 
but then it worked. Still running the search,11,211 vs 11,211 takes a while, so far so good though!

**Comment by seanrjohnson** [comment]
@Argusmocny The search ran for more than a week and then hit the same bug Cituglia40 mentioned above 
```
raceback (most recent call last):
  File "pLM-blast/pLM-BLAST/scripts/plmblast.py", line 219, in <module>
    query_result = aln.postprocess.prepare_output(args, rows, qid, query_seqs[qid], db_df)
  File "pLM-blast/pLM-BLAST/alntools/postprocess/format.py", line 61, in prepare_output
    raise KeyError(f'missing {col} column in input dataframe')
KeyError: 'missing id column in input dataframe'
```
😂

I'll run on a smaller dataset and see if I can debug it.

**Comment by Argusmocny** [comment]
We have released a big update in last days:
* now pre-screening is few order of magnitude faster, especially when query and database are large
* `db` and `query` both supports directory and file mode, so there is no need of calulating two times the same embedding when searching all vs all on single database
* type mismatches where fixed

additionally we added signal enhancement (described here: https://www.biorxiv.org/content/10.1101/2022.12.13.520313v1) , as `--enh` flag to `plmblast.py` script. Which could provide longer and more correct alignemnts.

PS. now `dbtofile.py` will be calculated on fly on first script launch for a given database (it needs to be in directory format)
